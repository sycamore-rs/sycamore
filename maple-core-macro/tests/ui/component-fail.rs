#![allow(non_snake_case)]

use maple_core::prelude::*;

fn Component() -> TemplateResult {
    template! {
        div
    }
}

fn compile_fail() {
    template! { UnknownComponent() };

    template! { Component };
    template! { Component(1) };
}

fn main() {}
