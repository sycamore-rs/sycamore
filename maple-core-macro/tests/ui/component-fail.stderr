error: unexpected end of input, expected parentheses
  --> $DIR/component-fail.rs:14:5
   |
14 |     template! { Component };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find function, tuple struct or tuple variant `UnknownComponent` in this scope
  --> $DIR/component-fail.rs:12:17
   |
12 |     template! { UnknownComponent() };
   |                 ^^^^^^^^^^^^^^^^ not found in this scope

error[E0061]: this function takes 0 arguments but 1 argument was supplied
  --> $DIR/component-fail.rs:15:17
   |
15 |     template! { Component(1) };
   |                 ^^^^^^^^^ - supplied 1 argument
   |                 |
   |                 expected 0 arguments
   |
note: function defined here
  --> $DIR/component-fail.rs:5:4
   |
5  | fn Component() -> TemplateResult {
   |    ^^^^^^^^^
